app/src/main/java/com/pavesid/androidacademy/ui/splash/SplashScreenFragment.kt

```diff
@@ -21,12 +21,14 @@ class SplashScreenFragment : Fragment(R.layout.fragment_splash_screen) {
 
     private var animateProgress = 0f
 
+    private val viewModel by lazy { ViewModelProvider(requireActivity()).get(MoviesViewModel::class.java) }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         animateProgress = savedInstanceState?.getFloat(PROGRESS) ?: animateProgress
 
-        ViewModelProvider(requireActivity()).get(MoviesViewModel::class.java)
+        viewModel.loadMovies()
```

Comment:
for what that? try to add delay(10000) to getMovies() method. Then your code will load movies two times. If you want to preload data at this place - you should check if data already loading in MoviesFragment.